src/transformations.jl

```diff
@@ -134,12 +133,24 @@ function fit(::Type{ZScoreTransform}, X::AbstractVector{<:Real};
         throw(DomainError(dims, "fit only accepts dims=1 over a vector. Try fit(t, x, dims=1)."))
     end
 
-    T = eltype(X)
-    m, s = mean_and_std(X)
-    return ZScoreTransform(1, dims, (center ? [m] : zeros(T, 0)),
-                                    (scale ? [s] : zeros(T, 0)))
+    return fit(ZScoreTransform, reshape(X, :, 1);dims=dims, center=center, scale=scale)
```

Comment:
The vector structure of the input input will be lost, and `transform` will return a 1-column matrix instead of a vector.

Reply:
Thanks for your review. I will fix this shortly. The issue is that when I stay closer to the original code, I got an inference error. I'll look for an alternate solution to work around that.

Reply:
It looks like `transform` still works as expected, i.e. it returns a `Vector`.
```julia
julia> X = rand(3)
3-element Vector{Float64}:
 0.0220367
 0.019156
 0.572394

julia> t = fit(ZScoreTransform, X)
ZScoreTransform{Float64, Vector{Float64}}(1, 1, [0.204529], [0.318584])

julia> transform(t, X)
3-element Vector{Float64}:
 -0.572823
 -0.581866
  1.15469

julia> Xt = transform(t, X)
3-element Vector{Float64}:
 -0.572823
 -0.581866
  1.15469

julia> reconstruct(t, Xt)
3-element Vector{Float64}:
 0.0220367
 0.019156
 0.572394
```

This is achieved through the `transform` [method](https://github.com/JuliaStats/StatsBase.jl/blob/master/src/transformations.jl#L23-L24) that is specialized for vectors: